commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java

```diff
@@ -223,6 +290,41 @@ public String getSelectedScreenNavigationCategoryKey() {
 				CATEGORY_KEY_COMMERCE_PAYMENT_METHOD_DETAILS);
 	}
 
+	public int getSubscriptionEntryCount(String engineKey)
+		throws PortalException {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)_renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		int activeSubscriptions = 0;
+
+		long groupId =
+			_commerceChannelLocalService.getCommerceChannelGroupIdBySiteGroupId(
+				themeDisplay.getSiteGroupId());
+
+		List<CommerceOrder> commerceOrders =
+			_commerceOrderLocalService.getCommerceOrders(groupId, engineKey);
+
+		for (CommerceOrder commerceOrder : commerceOrders) {
+			for (CommerceOrderItem commerceOrderItem :
+					commerceOrder.getCommerceOrderItems()) {
+
+				if (commerceOrderItem.isSubscription()) {
+					CommerceSubscriptionEntry commerceSubscriptionEntry =
+						_commerceSubscriptionEntryLocalService.
+							fetchCommerceSubscriptionEntryByCommerceOrderItemId(
+								commerceOrderItem.getCommerceOrderItemId());
+
+					if (commerceSubscriptionEntry != null) {
+						activeSubscriptions++;
+					}
+				}
+			}
+		}
```

Comment:
oof.... is there any cleaner way we can do this? This feels very.... brute-force. And could get really slow in a large system.

Reply:
if this is all for supplying the warning message with more context for the user, it may not be worth it.....

Reply:
Yeah this strikes me as something we should use the indexer for -- maybe even just add payment method key to the subscription entry document, and then query for a count that way.

Reply:
Fixed with finders